src/Draco.Compiler/Syntax/Lexer.cs

```diff
@@ -352,13 +352,33 @@ private IToken LexString()
         // Check for escape sequence
         if (ch == '\\')
         {
-            ++offset;
             // Count the number of required delimiters
             for (var i = 0; i < mode.ExtendedDelims; ++i)
             {
-                if (this.Peek(offset + i) != '#') goto not_escape_sequence;
+                if (this.Peek(offset + i + 1) != '#')
```

Comment:
I'd personally re-inline this goto so the if doesn't use braces anymore but that's up to you